components/apps.js

```diff
@@ -649,33 +647,137 @@ SteamUser.prototype.getOwnedDepots = function(excludeSharedLicenses) {
-	if (this.steamID.type != SteamID.Type.ANON_USER && !this.licenses) {
+SteamUser.prototype.getOwnedLicenses = function() {
+	if (!this.licenses) {
 		throw new Error("We don't have our license list yet.");
 	}
 
+	return this.licenses;
+}
+
+/**
+ * Returns an array of package IDs this account owns (different from owned licenses). The filter options
+ * only work, if enablePicsCache option is enabled and appOwnershipCached event has been emitted.
+ * @param {object|function} options - Options for what counts for ownership, or a custom filter function
+ * @returns {int[]}
+ */
+SteamUser.prototype.getOwnedPackages = function(options) {
+	// We're anonymous
 	if (this.steamID.type == SteamID.Type.ANON_USER) {
 		return [17906];
 	}
 
 	// We're an individual user
-	let packages = this.licenses;
-	if (excludeSharedLicenses) {
-		packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+	let packages = this.getOwnedLicenses();
+
+	// Adds support for the previous syntax
+	if (!options) {
+		// Slight deviation of old method: don't return expired licenses
+		return packages.filter(license => license.flags !== SteamUser.ELicenseFlags.Expired).map(license => license.package_id);
 	}
+	// If options is a boolean, we asssume it's excludeSharedLicenses
+	if (typeof options === 'boolean') {
+		let excludeSharedLicenses = options;
+		if (excludeSharedLicenses) {
+			packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+		}
+		// Slight deviation of old method: don't return expired licenses
+		return packages.filter(license => license.flags !== SteamUser.ELicenseFlags.Expired).map(license => license.package_id);
+	}
+
+	// Handle options
+	if (typeof options === 'object') {
+		const defaults = {
+			excludeFree: false, // By default, include free licenses (Sub 0 & FreeOnDemand & NoCost)
+			excludeShared: true, // By default, exclude shared licenses
+			excludeExpired: true // By default, exclude licenses that are going to expire (free weekends)
+		};
+		options = Object.assign(defaults, options);
+
+		// If there is nothing to filter, no pics cache needed
+		if (!options.excludeFree && !options.exludeShared && !options.excludeFree) {
+			return packages.map(license => license.package_id);
+		}
+	}
+
+	// From this point, we need pics cache to be enabled
+	this._ensurePicsCache();
+
+	// Determine the filter function
+	let packageFilter = () => true;
+	if (typeof options === 'function') {
+		packageFilter = options;
+	} else if (typeof options === 'object') {
+		packageFilter = (license) => {
+
+			// If expired, filter it out, regardless of the options
+			if (license.flags === SteamUser.ELicenseFlags.Expired) {
+				return false;
+			}
+
+			// If exclude shared licenses
+			if (options.excludeShared && license.owner_id !== this.steamID.accountid) {
+				return false;
+			}
+
+			let pkg = license.package_id;
+			if (!this.picsCache.packages[pkg]) {
```

Comment:
Changing this to `if (!this.picsCache.packages[pkg] || !this.picsCache.packages[pkg].packageinfo)` would remove the need for the next if statement.